docs/ARCHITECTURE.md

```diff
@@ -22,21 +22,46 @@ effective tool for finding and removing individual records within objects stored
 in S3 buckets. In order to achieve these goals the solution has adopted the
 following design principles:
 
-1. **Perform work in batches:** Since the time complexity of removing a single
-   vs multiple records in a single object is practically equal and it is common
-   for data owners to have the legal requirement of removing data within a given
-   _timeframe_, the solution is designed to allow the customer to "queue"
-   multiple matches to be removed in a single job.
-2. **Fail fast:** A deletion job takes place in two distinct phases: Find and
-   Forget. The Find phase queries the objects in your S3 data lakes to find any
-   objects which contain records where a specified column contains at least one
-   of record identifiers (known as **Match IDs**) in the deletion queue. If any
-   queries fail, the job will abandon as soon as possible and the Forget phase
-   will not take place. The Forget Phase takes the list of objects returned from
-   the Find phase, and deletes the relevant rows in only those objects.
-3. **Serverless:** Where possible, the solution only uses Serverless components.
-   All the components for Web UI, API and Deletion Jobs are Serverless (for more
-   information consult the [Cost Overview guide]).
+1. **Secure by design:** Every component is implemented with least privilege
+   access, encryption is performed at all layers at rest and in transit,
+   authentication is provided out of the box, expiration of logs is
+   configurable, and record identifiers (known as **Match IDs**) are
+   automatically obfuscated or irreversibly deleted as soon as possible when
+   persisting state.
+2. **Built to scale:** The system has been designed and tested for performance
```

Comment:
```suggestion
2. **Built to scale:** The system is designed and tested for performance
```

Reply:
also _designed and tested for performance and high availability with Data Lakes of the..._ - rereading this now I don't know if that actually makes sense. The scale of the data lake doesn't really impact availability? And the statement doesn't specify the standard of performance. I think we should be saying that _the system is designed and tested to support up to petabyte scale data lakes_ or something similar?